doc/advanced-topics.md

```diff
@@ -1,626 +0,0 @@
-
--   `addQueryMiddleware`: adds a queryContext middleware to the stack of middlewares. All the middlewares will be
-    applied to every call to the `query()` function.
-
--   `resetMiddlewares`: remove all the middlewares from the system.
-
-Usually, the full list of middlewares an IoT Agent will use would be added in the IoTAgent start sequence, so they
-should not change a lot during the IoT lifetime.
-
-#### Provided plugins
-
-The library provides some plugins out of the box, in the `dataPlugins` collection. In order to load any of them, just
-use the `addQueryMiddleware` and `addUpdateMiddleware` functions with the selected plugin, as in the example:
-
-```javascript
-var iotaLib = require('iotagent-node-lib');
-
-iotaLib.addUpdateMiddleware(iotaLib.dataPlugins.compressTimestamp.update);
-iotaLib.addQueryMiddleware(iotaLib.dataPlugins.compressTimestamp.query);
-```
-
-##### Timestamp Compression plugin (compressTimestamp)
-
-This plugins change all the timestamp attributes found in the entity, and all the timestamp metadata found in any
-attribute, from the basic complete calendar timestamp of the ISO8601 (e.g.: 20071103T131805) to the extended complete
-calendar timestamp (e.g.: +002007-11-03T13:18). The middleware expects to receive the basic format in updates and return
-it in queries (and viceversa, receive the extended one in queries and return it in updates).
-
-##### Attribute Alias plugin (attributeAlias)
-
-In the Device provision, an ID can be specified for each attribute, along with its name. The ID can be used then as the
-left part of a mapping from attribute names in the south bound to attribute names in the North Bound. If the ID and name
-attributes are used in this way, this plugin makes the translation from one to the other automatically.
-
-##### Event plugin (addEvents)
-
-This plugin allows for the creation of Event attributes, i.e.: attributes whose value will be the timestamp of its
-inclusion in the system, regardless of the value they carried in the Southbound API. If this plugin is active, all the
-events in the IoT Agent with the configured type name will be marked as events. The event name can be configured in the
-`config.eventType` attribute.
-
-##### Timestamp Processing Plugin (timestampProcess)
-
-This plugin processes the entity attributes looking for a `TimeInstant` attribute. If one is found, for NGSI v2, the
-plugin adds a `TimeInstant` attribute as metadata for every other attribute in the same request. With NGSI-LD, the
-Standard `observedAt` property-of-a-property is used instead.
-
-##### Expression Translation plugin (expressionTransformation)
-
-This plugin allows the devices and configurations that have defined expressions to generate their values from those
-expressions and the reported measure information.
-
-For further information on how the expressions work, refer to the
-[Expression Language Reference](expressionLanguage.md).
```

Comment:
This section was not moved to the `api.md` file since there is a complete file describing it that need to be included.